skimage/color/colorlabel.py

```diff
@@ -182,9 +182,15 @@ def _label2rgb_overlay(label, image=None, colors=None, alpha=0.3,
         # Opacity doesn't make sense if no image exists.
         alpha = 1
     else:
-        if not image.shape[:2] == label.shape:
+        if (image.shape[:label.ndim] != label.shape
+                or image.ndim > label.ndim + 1):
             raise ValueError("`image` and `label` must be the same shape")
 
+        if image.ndim == label.ndim + 1 and image.shape[-1] != 3:
```

Comment:
Shouldn't this be
```suggestion
        if image.ndim == label.ndim + 1 and image.shape[channel_axis] != 3:
```
?

Reply:
Nop, `channel_axis` has already been moved to last dimension in `label2rgb` :wink:

Reply:
:thinking: sorry I don't get it, since I'm seeing (unmodified by this PR):
https://github.com/scikit-image/scikit-image/blob/66979f2ecc82d09369e174083938e82157264341/skimage/color/colorlabel.py#L74-L76

Reply:
Ok, sorry, we are now inside of function `_label2rgb_overlay` indeed! Got mixed up expanding hidden code... :slightly_smiling_face: